client/core/core.go

```diff
@@ -1498,16 +1538,15 @@ func (c *Core) ReconfigureWallet(appPW []byte, assetID uint32, cfg map[string]st
 	if err != nil {
 		return newError(authErr, "ReconfigureWallet password error: %v", err)
 	}
-	c.walletMtx.Lock()
-	defer c.walletMtx.Unlock()
-	oldWallet, found := c.wallets[assetID]
```

Comment:
I'm not so sure this mutex change is desirable.  While there is the `(*Core).wallet` method that handles the locking, I think the reasoning for locking it up hear and keeping it locked until the new wallet is written into the map is to make the process atomic.  That is, we are guaranteed to be replacing oldWallet with the new one the old way, but now another goroutine could modify the map between this access and the write below.

Reply:
Reverted